app/controllers/amr_heatmap_controller.rb

```diff
@@ -68,169 +69,67 @@ def amr_counts
-      render(json: ontology) && return
-    end
-    ## Search the CARD OWL file
-    raw_owl_xml = card_owl.force_encoding("utf-8")
-    owl_doc = Nokogiri::XML(raw_owl_xml)
-    search_result = search_card_owl(gene_name, owl_doc)
-    unless search_result["error"].nil?
-      ontology["error"] = search_result["error"]
-      render(json: ontology) && return
-    end
+    ontology = {}
 
-    ontology_info = gather_ontology_information(search_result["matching_node"], owl_doc)
-    ontology_info.each do |key, value|
-      ontology[key] = value
-    end
-
-    ontology["publications"] = ontology["publications"].map do |pubmed_string|
-      get_publication_name(pubmed_string.split(":")[1])
+    ontology_json_key = fetch_current_ontology_file_key()
+    card_entry = fetch_ontology(ontology_json_key, gene_name)
+    if card_entry.key?("label")
+      ontology = card_entry
+      ontology["error"] = ""
+    else
+      ontology["error"] = "No data for #{gene_name}."
     end
-
     render json: ontology
   end
 
   private
 
-  def fetch_card_owl
-    card_owl = Rails.cache.fetch('card_owl_file', expires_in: 30.days) do
-      get_s3_file(S3_CARD_OWL)
-    end
-    return card_owl
+  def set_aws_client
+    @client = Aws::S3::Client.new(region: DEFAULT_S3_REGION)
   end
 
-  def search_card_owl(gene_name, xml_dom)
-    search_result = {}
-    err = "No match found for %s in the CARD Antibiotic Resistance Ontology."
-    owl_classes = xml_dom.xpath(".//owl:Class")
-    matching_entry = nil
-
-    # normalize gene name
-    alphanumeric_gene_name = gene_name.downcase.gsub(/\W/, '')
-    regex_for_gene = Regexp.new("\\b#{alphanumeric_gene_name}\\b")
-    # search through owl classes
-    owl_classes.each do |owl_class|
-      # first we check the label
-      label = owl_class.at_xpath("./rdfs:label").content
-      separated_label = label.downcase.tr('/', ' ')
-      alphanumeric_label = separated_label.gsub(/[^0-9a-z_\s]/, '')
-      if regex_for_gene.match?(alphanumeric_label)
-        matching_entry = owl_class
-        break
-      end
-
-      # then we check the synonyms
-      synonyms = owl_class.xpath("./oboInOwl:hasExactSynonym")
-      synonyms.each do |synonym|
-        separated_synonym = synonym.content.downcase.tr('/', ' ')
-        alphanumeric_synonym = separated_synonym.gsub(/[^0-9a-z_\s]/, '')
-        if regex_for_gene.match?(alphanumeric_synonym)
-          matching_entry = owl_class
-          break
-        end
-      end
-    end
-
-    # abort if no matching entry
-    if matching_entry.nil?
-      search_result["error"] = format(err, gene_name)
-    else
-      search_result["matching_node"] = matching_entry
-    end
-
-    return search_result
+  def fetch_current_ontology_file_key
+    ontology_folder = @client.list_objects_v2(bucket: S3_JSON_BUCKET,
+                                              prefix: S3_JSON_PREFIX).to_h
+    target_key = ontology_folder[:contents][-1][:key] # contents already sorted by key
```

Comment:
This is interesting. It looks like you automatically take the last file in the folder. An alternative approach would be to make this an environment variable (via Parameter store, I can show you how), or store it in the database somehow.

For our nt/nr indices, we have AlignmentConfig objects in our Rails model that store the paths to the index files. 

Will defer to others about whether it's worth it to take an alternative approach.

Reply:
Yeah, each time the rake task is run the json file is put in a folder `amr/ontology/YYYY-MM-DD/aro.json` so the latest run of the rake task will be the last key listed here. I like the simplicity of this way, but I don't mind changing it.

Reply:
it sounds fine to me to assume the last one (altho maybe I would sort it again just to be sure)

Reply:
Can you add a comment for this function explaining why the last key is the most recently generated file? (as you just explained in your Github comment)

Reply:
Done!